src/compiler/transformers/esnext.ts

```diff
@@ -212,6 +214,13 @@ namespace ts {
             return visitEachChild(node, noDestructuringValue ? visitorNoDestructuringValue : visitor, context);
         }
 
+        function visitCatchClause(node: CatchClause): CatchClause {
+            if (!node.variableDeclaration) {
+                return updateCatchClause(node, createVariableDeclaration(createTempVariable(/*recordTempVariable*/ undefined)), node.block);

```

Comment:
@weswigham is correct, you should recursively visit the block of the catch clause.